src/test/java/com/wikia/webdriver/common/testnglisteners/BrowserAndTestEventListener.java

```diff
@@ -0,0 +1,265 @@
+          driver.manage().addCookie(
+              new Cookie("tracking-opt-in-status", "rejected", cookieDomain, "/",
+                         cookieDate));
+        }
+      }
+
+      /**
+       * We want to disable sales pitch dialog for new potential contributors to avoid hiding other
+       * UI elements. see https://wikia-inc.atlassian.net/browse/CE-3768
+       */
+      if (TestContext.isFirstLoad() && "true"
+          .equals(Configuration.getDisableCommunityPageSalesPitchDialog())) {
+        driver.manage().addCookie(
+            new Cookie("cpBenefitsModalShown", "1", cookieDomain, "/", cookieDate));
+      }
+
+      if (TestContext.isFirstLoad() && "true".equals(Configuration.getMockAds())) {
+        driver.manage().addCookie(new Cookie("mock-ads", XMLReader.getValue("mock.ads_token"),
+                                             cookieDomain, "/", cookieDate));
+        Log.info(String.format("Adding moc-ads cookie with value: %s, and domain: %s",
+                               XMLReader.getValue("mock.ads_token"),
+                               String.format(".%s", Configuration.getEnvType()
+                                   .getWikiaDomain())));
+      }
+    }
+
+    if (TestContext.isFirstLoad()) {
+      User user = null;
+      TestContext.setFirstLoad(false);
+
+      if (declaringClass.isAnnotationPresent(Execute.class)) {
+        user = declaringClass.getAnnotation(Execute.class).asUser();
+      }
+
+      if (method.isAnnotationPresent(Execute.class)) {
+        user = method.getAnnotation(Execute.class).asUser();
+      }
+
+      if (user != null && user != User.ANONYMOUS) {
+        // log in, make sure user is logged in and flow is on the requested url
+        new WikiBasePageObject().loginAs(user);
+      }
+      NetworkTrafficInterceptor
+          networkTrafficInterceptor =
+          DriverProvider.getActiveDriver().getProxy();
+      if (networkTrafficInterceptor != null && Configuration.getForceHttps()) {
+        networkTrafficInterceptor.startIntercepting();
+      }
+    }
+
+    Log.logJSError();
+  }
+
+  @Override
+  public void beforeFindBy(By by, WebElement element, WebDriver driver) {
+    lastFindBy = by;
+    SelectorStack.write(by);
+    if (element != null) {
+      SelectorStack.contextWrite();
+    }
+  }
+
+  @Override
+  public void beforeClickOn(WebElement element, WebDriver driver) {
+    Log.logJSError();
+  }
+
+  @Override
+  public void afterClickOn(WebElement element, WebDriver driver) {
+    Log.info("click", lastFindBy.toString());
+  }
+
+  @Override
+  public void afterChangeValueOf(WebElement element, WebDriver driver, CharSequence[] keysToSend) {
+    Log.info("ChangeValueOfField", lastFindBy.toString());
+  }
+
+  @Override
+  public void onTestStart(ITestResult result) {
+    Log.clearLogStack();
+    String testName = result.getName();
+    String className = result.getTestClass().getName();
+    System.out.println(className + " " + testName);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS106)